File: src/AAS.Domain/EmailVerifications/EmailVerification.cs
using AAS.Tools.Types.IDs;

namespace AAS.Domain.EmailVerifications;

public class EmailVerification
{
    public ID UserId { get; }
    public String Token { get; }
    public Boolean IsVerified { get; }

    public EmailVerification(ID userId, string token, bool isVerified = false)
    {
        UserId = userId;
        Token = token;
        IsVerified = isVerified;
    }
}

File: src/AAS.BackOffice/client_app/src/domain/emailVerifications/emailVerification.ts
export class EmailVerification {
    constructor(
        public readonly userId: string,
        public readonly token: string,
        public readonly isVerified: boolean
    ) { }
}

export function mapToEmailVerification(value: any): EmailVerification {
    return new EmailVerification(value.userId, value.token, value.isVerified);
}